android/src/main/java/org/conscrypt/Platform.java

```diff
@@ -849,24 +849,18 @@ public static String oidToAlgorithmName(String oid) {
         return oid;
     }
 
-    /*
-     * Pre-Java 8 backward compatibility.
+    /**
+     * Provides extended capabilities for the session if supported by the platform.
      */
-
-    public static SSLSession wrapSSLSession(ActiveSession sslSession) {
-        if (Build.VERSION.SDK_INT <= 23) {
-            return sslSession;
+    public static SSLSession wrapSSLSession(ConscryptSession sslSession) {
+        if (Build.VERSION.SDK_INT >= 24) {
+            return new Java8ExtendedSSLSession(sslSession);
         }
-
-        return new Java7SessionWrapper(sslSession);
-    }
-
-    public static SSLSession unwrapSSLSession(SSLSession sslSession) {
-        if (Build.VERSION.SDK_INT <= 23) {
-            return sslSession;
+        if (Build.VERSION.SDK_INT >= 19) {
```

Comment:
flooey@ one last thing ... is this the correct API version for Java 7?

Reply:
I believe so, yep.

Reply:
I don't think this is true. 19 is the right SDK version for using commonly used Java 7 language features (try-with-resources and the like). It does not guarantee the presence of all Java 7 APIs. If you're doing this for ExtendedSSLSession, i don't believe we had it until API 24. flooey@ : did you confirm we had this class in 19 ?